app/models/concerns/katalyst/content/active_storage_extensions/has_one_image_attached.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module Katalyst
+  module Content
+    module ActiveStorageExtensions
+      module HasOneImageAttached
+        extend ActiveSupport::Concern
+
+        class_methods do
+          # rubocop:disable Naming/PredicateName
+          # rubocop:disable Style/DocumentDynamicEvalDefinition
+          def has_one_image_attached(name)
+            class_eval <<-CODE, __FILE__, __LINE__ + 1
+          has_one_attached :#{name}
+
+          validates :#{name},
+                content_type: Katalyst::Content.config.supported_image_mime_types,
+                size:         {#{' '}
+                                less_than: Katalyst::Content.config.supported_max_image_size.megabytes,#{' '}
+                                message: "must be less than #{Katalyst::Content.config.supported_max_image_size} MB"#{' '}
+                }
+
+          def destroy_#{name}; end
+
+          def destroy_#{name}=(value)
+            self.#{name} = nil if value.present?
+          end
```

Comment:
check if this is needed.

should be able to just do `#{name}=nil` without needing a destroy method

Reply:
Removed